### PR TITLE
Archive.containsGroupWithTitle

### PR DESCRIPTION
**Pull request by Jameskmonger**
This PR adds the hasGroup function to the Archive class. It allows for the user to see if a group with the given title exists in the archive.

### CODE REVIEW

File: source/classes/Archive.js [hunk 1]
```diff
@@ -66,6 +66,17 @@
 	};
 
 	/**
+	 * Whether or not this archive has a group with the given title.
+	 * @param {String} The group's title
+	 * @returns {true|false}
+	 */
+	Archive.prototype.hasGroup = function(groupTitle) {
```
Comment (perry-mitchell) [inline source/classes/Archive.js]: Perhaps using the name `containsGroupWithTitle` would be better suited. My reasoning:
 * containsGroup (vs hasGroup) I feel better reflects the recursiveness of the method (hasGroup sounds shallow)
 * I'd like to use `containsGroup` for searching with an actual group

What do you think?

Comment (Jameskmonger) [inline source/classes/Archive.js]: All sounds good to me @perry-mitchell. I'll get on that now.

**Comment by Jameskmonger** [comment]
@perry-mitchell The build passed on all instances except one, where it failed due to the issue explained in #39

**Comment by perry-mitchell** [comment]
@Jameskmonger Yeah that one's a pain. Rerunning...

**Comment by perry-mitchell** [comment]
Thanks for this!

**Comment by Jameskmonger** [comment]
No problem @perry-mitchell! Will have another PR for you later today or tomorrow! :smile:

**Pull request merged**